src/endpoint/endpoint.js

```diff
@@ -111,18 +118,75 @@ async function run_server(options) {
         dbg.log0('S3 server started successfully');
         dbg.log0('Configured Virtual Hosts:', VIRTUAL_HOSTS);
 
+        // Start a monitor to send periodic endpoint reports about endpoint usage.
+        start_monitor();
+
     } catch (err) {
         handle_server_error(err);
     }
 }
 
+async function start_monitor() {
+    await promise_utils.wait_until(() =>
+        system_store.is_finished_initial_load
+    );
+    const system = system_store.data.systems[0];
+    const auth_token = auth_server.make_auth_token({
+        system_id: system._id,
+        account_id: system.owner._id,
+        role: 'admin'
+    });
+
+    let start_time = process.hrtime.bigint() / 1000n;
+    let base_cpu_time = process.cpuUsage();
+    for (;;) {
+        try {
+            await promise_utils.delay_unblocking(ENDPOINT_MONITOR_INTERVAL);
+            const end_time = process.hrtime.bigint() / 1000n;
+            const cpu_time = process.cpuUsage();
```

Comment:
I think that you can get the elapsed CPU time by passing `base_cpu_time` to `process.cpuUsage`

```suggestion
            const cpu_time = process.cpuUsage(base_cpu_time);
            const cpu_usage = (cpu_time.system + cpu_time.user) / Number((end_time - start_time));
```

Reply:
I am saving CPU_time into base_cpu_time for the next iteration, so cpu_time cannot be the delta